crates/bevy_mod_scripting_core/src/reflection_extensions.rs

```diff
@@ -424,11 +425,19 @@ impl<T: PartialReflect + ?Sized> PartialReflectExt for T {
     fn from_reflect_or_clone(
         reflect: &dyn PartialReflect,
         world: WorldGuard,
-    ) -> Box<dyn PartialReflect> {
+    ) -> Result<Box<dyn PartialReflect>, InteropError> {
         // try from reflect
         match <dyn PartialReflect>::from_reflect(reflect, world.clone()) {
-            Ok(v) => v.into_partial_reflect(),
-            Err(_) => reflect.clone_value(),
+            Ok(v) => Ok(v.into_partial_reflect()),
+            Err(_) => reflect
```

Comment:
I believe this is what we ultimately wanted this function to do, I can't remember if allowing non-fully-reflect values to be returned here was a convenience thing and it just never happened or if it was used for something at specific at some point, either way, if it causes trouble we can have a look